extensions/webdav/src/org/exist/webdav/ExistCollection.java

```diff
@@ -443,17 +443,18 @@ void resourceCopyMove(XmldbURI destCollectionUri, String newName, Mode mode) thr
         }
 
         Collection srcCollection = null;
-        final LockMode srcCollectionLockMode = (mode == Mode.COPY ? LockMode.READ_LOCK : LockMode.WRITE_LOCK);
+        // This class contains already the URI of the resource that shall be moved/copied
+        final XmldbURI srcCollectionUri = xmldbUri;
+        // use WRITE_LOCK if moving or if src and dest collection are the same
```

Comment:
though not needed, you might want for readability write this as

`LockMode srcCollectionLockMode = (mode == Mode.MOVE
 +                || destCollectionUri.equals(srcCollectionUri)) ? LockMode.WRITE_LOCK : LockMode.READ_LOCK;`